### PR TITLE
[WIP] Adds new ship spawning verb

### LINKED ISSUE #1127: Shuttles spawned via debug place, and uploaded, do not generate an overmap token[BUG]
## Round ID:
[890](https://shiptest.ga/stats/rounds/890)

## Testmerges:
[#132: 'Readds Explosive Decompression' by MarkSuckerberg at commit 715c79f1bd](https://github.com/Shiptest-SS13/Shiptest/pull/132)
[#941: 'Trims down the """Economy"""' by MarkSuckerberg at commit bba9fc235b](https://github.com/Shiptest-SS13/Shiptest/pull/941)
[#1060: 'Adds the Caravan-Class' by AquillaF at commit b178128b7e](https://github.com/Shiptest-SS13/Shiptest/pull/1060)
[#1061: 'IRMG Colossus-class Armored Frigate' by Apogee-dev at commit 11622615c4](https://github.com/Shiptest-SS13/Shiptest/pull/1061)
[#1025: 'New Ship - Masinyane' by Vexylius at commit 329ba56fc3](https://github.com/Shiptest-SS13/Shiptest/pull/1025)
[#1085: 'Rube-Goldberg Class Engineering Project (Gone OSHA)' by SweetBlueSylveon at commit 2758da24a6](https://github.com/Shiptest-SS13/Shiptest/pull/1085)
[#1007: 'Keppification' by MarkSuckerberg at commit c8cd6f85b9](https://github.com/Shiptest-SS13/Shiptest/pull/1007)
[#1058: '[TM-ONLY] Revert "speech controllers and hearers_in_view improvements"' by ZephyrTFA at commit ff8e5562a3](https://github.com/Shiptest-SS13/Shiptest/pull/1058)
[#1001: 'Adds the options for IPCs to have their brains as MMI's' by Bokkiewokkie at commit ebee854692](https://github.com/Shiptest-SS13/Shiptest/pull/1001)
[#1096: 'Midway Remaster' by PrefabQuasar at commit 9f96c975e3](https://github.com/Shiptest-SS13/Shiptest/pull/1096)
[#1038: 'Fixes annoying issues + Remaps a jungle ruin' by Zevotech at commit a909267502](https://github.com/Shiptest-SS13/Shiptest/pull/1038)
[#945: 'The MedTech™ chemical manufacturing plant ruin ' by disappointedButNotSuprised at commit 71fbef1c9a](https://github.com/Shiptest-SS13/Shiptest/pull/945)
[#1070: 'Boyardee-B update' by bluezorua at commit 256692ee9e](https://github.com/Shiptest-SS13/Shiptest/pull/1070)
[#1108: '[DNM] Update smash.ogg' by tiramisuapimancer at commit 14c3ff6942](https://github.com/Shiptest-SS13/Shiptest/pull/1108)
[#1120: 'Remaps jungle_seedling.dmm' by Zevotech at commit e1b4ad422f](https://github.com/Shiptest-SS13/Shiptest/pull/1120)

## Reproduction:
As an admin, upload a map template shuttle, define it as shuttle, place it. Ship won't work. After extensive testing with VV, the ship does not seem to get an overmap token generated when spawned, causing everything to fail. You can kind of get around this by spawning a shuttle with the manip, and stripping it of its token, and vving it onto the linked_ship of the docking port and helm console at minimum. You will also need to set the tokens set docking port to the one you spawned in.

### PR DESCRIPTION
**Pull request by swifterjack**
## About The Pull Request
Fixes #1127 by adding a verb that loads any templates you feed into it as a shiptest and generates an overmap token.

Adds new verb, ship_template_load / "Load Template As Ship" which creates an overmap token and ship datum for the selected template and loads it like it was spawned with the shuttle manip. The way I do it is a bit jank, so please tell me if there is a better way.
## Why It's Good For The Game
Fixes bug, lets us spawn out of repo maps. Jobs still dont work, but thats a bit beyond my comprehension right now.

## Changelog
:cl:
add: Added new spawning verb
admin: New verb in debug menu
fix: template spawned ships fly again
/:cl:

### CODE REVIEW

**Comment by swifterjack** [comment]
Note: I have not yet tested the effects of selecting a non-ship template, but i feel like admins can be trusted to not try to spawn lavaland_surface_syndicate_base1.dmm as a shiptest

**Comment by Cenrus** [comment]
It might be better to add OM token generation to the shuttle manipulator, it's a really useful tool and even lets you preview the shuttle before you spawn it in.

**Comment by Fikou** [comment]
> Note: I have not yet tested the effects of selecting a non-ship template, but i feel like admins can be trusted to not try to spawn lavaland_surface_syndicate_base1.dmm as a shiptest

test it please

File: code/modules/admin/verbs/map_template_loadverb.dm [hunk 1]
```diff
@@ -26,6 +26,21 @@
...
+
+	var/datum/map_template/template
+
+	var/map = input(src, "Choose a Map Template to spawn as a ship.","Place Map Template") as null|anything in sortList(SSmapping.map_templates)
+	if(!map)
+		return
+	template = SSmapping.map_templates[map]
+
+	if(alert(src,"Confirm Placement.","Ship Confirm","Yes","No") == "Yes")
+		var/datum/overmap/ship/controlled/target = new(SSovermap.get_unused_overmap_square(), template)
```
Comment (MarkSuckerberg) [inline code/modules/admin/verbs/map_template_loadverb.dm:41]:
```suggestion
		new /datum/overmap/ship/controlled(SSovermap.get_unused_overmap_square(), template)
```

File: code/modules/admin/verbs/map_template_loadverb.dm [hunk 2]
```diff
@@ -26,6 +26,21 @@
...
+/client/proc/ship_template_load()
+	set category = "Debug"
+	set name = "Load Template As Ship"
+
+	var/datum/map_template/template
+
+	var/map = input(src, "Choose a Map Template to spawn as a ship.","Place Map Template") as null|anything in sortList(SSmapping.map_templates)
+	if(!map)
+		return
+	template = SSmapping.map_templates[map]
```
Comment (MarkSuckerberg) [inline code/modules/admin/verbs/map_template_loadverb.dm:38]:
```suggestion
	var/datum/map_template/template = SSmapping.map_templates[input(src, "Choose a Map Template to spawn as a ship.","Place Map Template") as null|anything in sortList(SSmapping.map_templates)]
```
this one's kind of up to choice, just putting it here if you want it to be one line, don't feel forced to do this

**Review by MarkSuckerberg — CHANGES_REQUESTED** [review]

**Comment by ZephyrTFA** [comment]
Sad

**Pull request closed without merging**